test/e2e/storage/testsuites/multiattach.go

```diff
@@ -0,0 +1,271 @@
+		// resource for 2nd volume
+		resource2 *genericVolumeTestResource
+	}
+	var (
+		dInfo = driver.GetDriverInfo()
+		l     local
+	)
+
+	BeforeEach(func() {
+		// Check preconditions.
+		if pattern.VolType != testpatterns.DynamicPV {
+			framework.Skipf("Suite %q does not support %v", t.tsInfo.name, pattern.VolType)
+		}
+		_, ok := driver.(DynamicPVTestDriver)
+		if !ok {
+			framework.Skipf("Driver %s doesn't support %v -- skipping", dInfo.Name, pattern.VolType)
+		}
+		if pattern.VolMode == v1.PersistentVolumeBlock && !dInfo.Capabilities[CapBlock] {
+			framework.Skipf("Driver %s doesn't support %v -- skipping", dInfo.Name, pattern.VolMode)
+		}
+	})
+
+	// This intentionally comes after checking the preconditions because it
+	// registers its own BeforeEach which creates the namespace. Beware that it
+	// also registers an AfterEach which renders f unusable. Any code using
+	// f must run inside an It or Context callback.
+	f := framework.NewDefaultFramework("multiattach")
+
+	init := func() {
+		l = local{}
+		l.ns = f.Namespace
+		l.cs = f.ClientSet
+
+		// Now do the more expensive test initialization.
+		l.config, l.testCleanup = driver.PrepareTest(f)
+
+		l.resource1 = createGenericVolumeTestResource(driver, l.config, pattern)
+		l.resource2 = &genericVolumeTestResource{
+			driver:  driver,
+			config:  l.config,
+			pattern: pattern,
+		}
+	}
+
+	cleanup := func() {
+		if l.resource1 != nil {
+			l.resource1.cleanupResource()
+			l.resource1 = nil
+		}
+
+		if l.resource2 != nil {
+			l.resource2.cleanupResource()
+			l.resource2 = nil
+		}
+
+		if l.testCleanup != nil {
+			l.testCleanup()
+			l.testCleanup = nil
+		}
+	}
+
+	testAttachTwoVolumesToOnePod := func(l local) {
+		var err error
+
+		l.resource2.sc = l.resource1.sc
+
+		if dDriver, ok := l.resource2.driver.(DynamicPVTestDriver); ok {
+			claimSize := dDriver.GetClaimSize()
+			l.resource2.pvc = getClaim(claimSize, l.ns.Name)
+			l.resource2.pvc.Spec.StorageClassName = &l.resource2.sc.Name
+			l.resource2.pvc.Spec.VolumeMode = &l.resource2.pattern.VolMode
+		} else {
+			framework.Skipf("Driver %q does not define Dynamic Provision StorageClass - skipping", l.resource2.driver.GetDriverInfo().Name)
+		}
+
+		By("Creating pv and pvc")
+		l.resource2.pvc, err = l.cs.CoreV1().PersistentVolumeClaims(l.ns.Name).Create(l.resource2.pvc)
+		Expect(err).NotTo(HaveOccurred())
+
+		err = framework.WaitForPersistentVolumeClaimPhase(v1.ClaimBound, l.cs, l.resource2.pvc.Namespace, l.resource2.pvc.Name, framework.Poll, framework.ClaimProvisionTimeout)
+		Expect(err).NotTo(HaveOccurred())
+
+		l.resource2.pvc, err = l.cs.CoreV1().PersistentVolumeClaims(l.resource2.pvc.Namespace).Get(l.resource2.pvc.Name, metav1.GetOptions{})
+		Expect(err).NotTo(HaveOccurred())
+
+		l.resource2.pv, err = l.cs.CoreV1().PersistentVolumes().Get(l.resource2.pvc.Spec.VolumeName, metav1.GetOptions{})
+		Expect(err).NotTo(HaveOccurred())
+
+		By("Creating pod with the two volumes with the same volume mode")
```

Comment:
the volume modes might be different

Reply:
Thank you for pointing it out! I just copy & past when refactoring locally.